### DISCUSSION TITLE
Exporting tokens references to figma layers

### DISCUSSION THREAD

**Discussion by jiricerhan** [six7/figma-tokens#507] [Q&A] [answered]
Hi, is there a way to export token references to Figma layers? I am trying to use your plugin's data to generate some code and this feature would be useful.

Something like:
```
[
{
  "id": "...",
  "type": "COMPONENT",
   ...
  "tokens": {
     "cornerRadius" : "global.small-radius",
      ...
  }
} ,
]                     
```

Is it possible?
Thanks a lot.

**Comment by six7** [comment, marked as answer]
The plugin currently stores all applied data on the sharedPluginData space `tokens`. If you were using the API you could extract these. There's quite a few properties you'd need to iterate on, but you can use `getSharedPluginDataKeys("tokens")` to find out which keys are set on this layer, and then find the value using `getSharedPluginData("tokens", key)`

Can you tell me a bit more about your use case? Would you like to find these via Figma's REST API?

**Reply by jiricerhan** [reply]
Thanks for the quick answer. That is exactly what I was looking for. 

I'm not that much familiar with Figma's API yet. I found information that plugin's data are private and overlooked this shared sharedPluginData thing. I thought it wasn't possible to get this from the API.  Sorry to bother you.

Thanks a lot again, for your help and this amazing plugin.
